src/core/font_renderer.js

```diff
@@ -25,19 +25,34 @@ import { getGlyphsUnicode } from "./glyphlist.js";
 import { StandardEncoding } from "./encodings.js";
 import { Stream } from "./stream.js";
 
-function getLong(data, offset) {
+// TODO: use DataView and its methods.
+
+function getUint32(data, offset) {
   return (
-    (data[offset] << 24) |
-    (data[offset + 1] << 16) |
-    (data[offset + 2] << 8) |
-    data[offset + 3]
+    ((data[offset] << 24) |
+      (data[offset + 1] << 16) |
+      (data[offset + 2] << 8) |
+      data[offset + 3]) >>>
+    0
   );
 }
 
-function getUshort(data, offset) {
+function getUint16(data, offset) {
   return (data[offset] << 8) | data[offset + 1];
 }
 
+function getInt16(data, offset) {
+  return ((data[offset] << 24) | (data[offset + 1] << 16)) >> 16;
+}
+
+function getInt8(data, offset) {
+  return (data[offset] << 24) >> 24;
+}
+
+function getFloat214(data, offset) {
+  return getInt16(data, offset) / 16384.0;
```

Comment:
Is the decimal really necessary here, since I believe that JS engines will just disregard it?